Console-2857: port protractor Overview list tests

florkbr opened the pull request:
Migrates overview list tests from protractor to cypress.

![image](https://user-images.githubusercontent.com/21317056/118729689-5a579000-b804-11eb-80db-92c08bbcbc0c.png)

https://issues.redhat.com/browse/CONSOLE-2857

dtaylor113 commented on frontend/packages/integration-tests-cypress/tests/app/overview.spec.ts:
```diff
@@ -0,0 +1,82 @@
+    checkErrors();
+  });
+
+  before(() => {
```
Small nit.  typically it goes
- before (all)
- beforeEach
- afterEach
- after (all)

Just for readability.  -thanks

florkbr replied:
@dtaylor113 done

dtaylor113 commented on frontend/packages/integration-tests-cypress/tests/app/overview.spec.ts:
```diff
@@ -0,0 +1,82 @@
+        });
+      });
+
+      it(`displays a ${kindModel.id} in the overview list page`, () => {
```
`it`'s typically start with a `cy.visit` or navigation to somewhere.  The helps test independance and when tests need to be retried.  I suggest moving `cy.visit('/k8s/cluster/projects/${testName}/workloads');` from the `before()` to the first command of the `it` test.

florkbr replied:
@dtaylor113 done

dtaylor113 commented on frontend/packages/integration-tests-cypress/tests/app/overview.spec.ts:
```diff
@@ -0,0 +1,82 @@
+      });
+
+      /* Disabling for now due to flake https://jira.coreos.com/browse/CONSOLE-1298
+      xit(`CONSOLE-1298 - shows ${kindModel.id} details sidebar when item is clicked`, () => {
```
Hi, so [CONSOLE-1298](https://issues.redhat.com/browse/CONSOLE-1298) is "Test flake: Visiting Overview shows details sidebar when item is clicked".   I recommend recreating this test in Cypress and see if it is still an issue.  Cypress in general has better 'waiting' conditions and it would be good to verify this behavior when clicking on an item.

florkbr replied:
@dtaylor113 done!

dtaylor113 commented on frontend/packages/integration-tests-cypress/views/overview.ts:
```diff
@@ -0,0 +1,14 @@
+export const overviewPage = {
+  projectOverviewIsVisible: () => {
```
Recommend `projectOverviewShouldBeVisible`  for consistant naming (as the assertion ends in `.should('be.visible')`)
idk, this is kinda how it has evolved.  Personally, I'm ok with 'Is' so feel free to push back  :-)

florkbr replied:
@dtaylor113 done!

dtaylor113 commented on frontend/packages/integration-tests-cypress/views/overview.ts:
```diff
@@ -0,0 +1,14 @@
+  projectOverviewIsVisible: () => {
+    cy.get(`.odc-topology-list-view`).should('be.visible');
+  },
+  itemsAreVisible: () => {
```
Recommend `itemsShouldBeVisible` for consistency

florkbr replied:
Done

dtaylor113 commented on frontend/packages/integration-tests-cypress/views/overview.ts:
```diff
@@ -0,0 +1,14 @@
+  groupLabelItemPresent: (kind) => {
+    cy.get('.odc-topology-list-view__kind-label').contains(kind.label);
+  },
+  projectOverviewListItemPresent: (name) => {
```
`groupLabelItemContains`
`projectOverviewListItemContains`

Also, how do you feel about #8878?  I updated for consistency and readability but I think it's somewhat redundant to have `.contains('..').should('exist')`

florkbr replied:
@dtaylor113 done; and agreed `.contains()` followed by `should('exist')` seems redundant.

dtaylor113 commented on frontend/packages/integration-tests-cypress/tests/app/overview.spec.ts:
```diff
@@ -0,0 +1,82 @@
+      });
+
+      it(`displays a ${kindModel.id} in the overview list page`, () => {
+        overviewPage.projectOverviewIsVisible();
```
I'm a little confused, when I goto `http://localhost:9000/k8s/cluster/projects/daves-proj/workloads` it appends `...?view=graph`  hmm...I see it actually appends `?view=graph` or `?view=list` depending on where user last visited, I guess it defaults to `?view=list` on new CI clusters?  

I recommend doing `cy.visit('/k8s/cluster/projects/${testName}/workloads?view=list');`  since these tests assume a topology LIST view, not the graphical topology editor.

florkbr replied:
@dtaylor113 done

dtaylor113 reviewed (changes requested):
Thanks @florkbr, looks good!  Just a few small comments.

florkbr commented:
@dtaylor113 please re-review. I've addressed your comments. I've included a screenshot of the tests passing locally below with the latest changes:
<img width="1441" alt="Screen Shot 2021-05-20 at 3 36 50 PM" src="https://user-images.githubusercontent.com/21317056/119038583-484f2c00-b981-11eb-8cd2-2368c12c55af.png">

dtaylor113 commented on frontend/packages/integration-tests-cypress/tests/app/overview.spec.ts:
```diff
@@ -0,0 +1,79 @@
+        overviewPage.projectOverviewListItemContains(resourceName);
+      });
+
+      it(`CONSOLE-1298 - shows ${kindModel.id} details sidebar when item is clicked`, () => {
```
Hi @florkbr looks good.  You can remove "CONSOLE-1298 -" from this test title.
I'll approve after this small change is done  -thanks!

florkbr replied:
@dtaylor113 done!

dtaylor113 reviewed (approved):
/lgtm

florkbr commented:
/assign @jhadvig

jhadvig commented:
/approve
